jsonapi/marshal.go

```diff
@@ -285,67 +287,68 @@ func getStructRelationships(relationer MarshalLinkedRelations, information Serve
 		relationships[name] = relationship
 	}
 
-	return &relationships
+	return relationships
 }
 
 // helper method to generate URL fields for `links`
 func getLinksForServerInformation(relationer MarshalLinkedRelations, name string, information ServerInformation) *Links {
-	links := &Links{}
-
-	if information != serverInformationNil {
-		prefix := strings.Trim(information.GetBaseURL(), "/")
-		namespace := strings.Trim(information.GetPrefix(), "/")
-		structType := getStructType(relationer)
+	if information == nil {
```

Comment:
As ServerInformation is an interface, the parameter information should either point to an implementor or be nil. I don't see the need for `serverInformationNil`. Maybe I'm missing something here.